src/isar/state_machine/state_machine.py

```diff
@@ -182,13 +160,19 @@ def __init__(
                 {
                     "trigger": "task_finished",
                     "source": self.monitor_state,
-                    "dest": self.initiate_state,
-                    "before": self._task_finished,
+                    "dest": self.monitor_state,
+                    "conditions": self._start_task,
+                },
+                {
```

Comment:
This transition is basically the "else" which is run if the conditional above returns false. In this case, if it fails to continue the mission (for instance if there are no more tasks) it will go to the idle state and run the "_full_mission_finished" cleanup function.

Reply:
The same is true for the transition from idle to idle. This is for when we fail to start a mission.